src/main/java/gmailgdogra/SwipeProcessor.java

```diff
@@ -39,56 +34,51 @@
 
     }
 
-    public SwipeProcessor(List<SwipeRecord> swipeRecords) {
-        this.swipeRecords = swipeRecords;
+    public static List<SwipeRecord> getOutputDataFrom(List<SwipeRecord> allSwipes) {
+
+        List<SwipeRecord> outputData = new ArrayList<>();
+        Set<Officer> officers = UserInput.getOfficers(allSwipes);
+        for (Officer officer : officers) {
+            if (officer.getLocation() != null) {
+                outputData.add(getFirstSwipeIn(officer, allSwipes)); // swipe in row
+                outputData.add(getLastSwipeOut(officer, allSwipes)); // swipe out row
+            }
+        }
+        return outputData;
     }
 
-    public SwipeRecord getFirstSwipeIn(Officer officer) {
-        var allInOutSwipesOfAnOfficer = getAllInOutSwipesOfAnOfficer(officer);
-        return allInOutSwipesOfAnOfficer.stream()
-                .filter(swipeRecord -> allSwipeInDevices.get(officer.getLocation()).contains(swipeRecord.getDeviceName()))
-                .filter(swipeRecord -> {
+    public static SwipeRecord getFirstSwipeIn(Officer officer, List<SwipeRecord> allSwipes) {
+        Location loc = officer.getLocation();
+        return allSwipes.stream()
+                .filter(swipe -> officer.getFullName().equals(swipe.getFullName()))
+                .filter(swipe -> allSwipeInDevices.get(loc).contains(swipe.getDeviceName()))
+                .filter(swipe -> {
                     if (officer.isDayShift()) {
-                        return swipeRecord.getSwipeTime().getHour() <= 12; // touch-in is before 12 noon
+                        return swipe.getSwipeTime().getHour() <= 12; // touch-in is before 12 noon
```

Comment:
I'm not sure about your business logic. In your case, 12:30 will be treated as "before 12 noon". Is it correct?

Reply:
Let me explain

I am trying to find if it is a day shift(0600 - 1800) or a night shift(1800-0600). Almost every time touch-In happen between 0500-0700 for morning shift, and 1700-1900 for night shift. for clarity I am going to change the timing to most likely scenario, i.e. 0500-0800, 1700-2000

Reply:
updated the comments for more clarity


```
  if (officer.isDayShift()) {
                        return swipe.getSwipeTime().getHour() <= 12; // touch-in is before 12:59 afternoon
                    } else {
                        return swipe.getSwipeTime().getHour() > 12; // touch-in on or after 13:00
                    }

```